crypto/asn1/tasn_enc.c

```diff
@@ -347,6 +350,10 @@ static int asn1_template_ex_i2d(const ASN1_VALUE **pval, unsigned char **out,
         i = ASN1_item_ex_i2d(pval, NULL, ASN1_ITEM_ptr(tt->item), -1, iclass);
         if (!i)
             return 0;
+        if (i == 0 && (tt->flags & ASN1_TFLG_OPTIONAL) == 0) {
```

Comment:
I believe this block is unreachable because the line immediately after it would have returned zero first. It seems:

1. They should be swapped.
2. This call is missing a check for negative values.